src/engine/strat_engine/engine.rs

```diff
@@ -124,6 +140,58 @@ impl Engine for StratEngine {
         Ok(uuid)
     }
 
+    /// Evaluate a device node & devicemapper::Device to see if it's a valid
+    /// stratis device.  If all the devices are present in the pool and the pool isn't already
+    /// up and running, it will get setup and the pool uuid will be returned.
+    ///
+    /// Returns an error if the status of the block device can not be evaluated.
+    fn block_evaluate(&mut self,
+                      dev_node: PathBuf,
+                      device: Device)
+                      -> EngineResult<Option<PoolUuid>> {
+        if let Some(pool_uuid) = is_stratis_device(&dev_node)? {
+            if self.pools.contains_uuid(pool_uuid) {
+                warn!("Pool {} is already known, ignoring!", pool_uuid);
+                Ok(None)
+            } else {
+                self.incomplete_pools
+                    .entry(pool_uuid)
+                    .or_insert_with(HashMap::new)
+                    .insert(device, dev_node);
+
+                let devices = self.incomplete_pools
+                    .get(&pool_uuid)
+                    .expect("We previously inserted pool uuid!")
+                    .clone();
+
+                let rc = if let Ok(pool) = StratPool::setup(pool_uuid, &devices) {
```

Comment:
I think we're going to discuss the problem of duplicate names in team meeting, so there's no reason to try to implement my suggestion before that.

However, I grant your previous point that setting up a device that you can't use just to tear it down again is not so great. My suggestion would be to pull the call to ```get_metadata``` out of ```StratPool::setup()``` and invoke it here. That way you can check whether there is a name conflict _before_ the pool is setup.

As far as I'm concerned, leaving the list of devs stuck in ```incomplete_pools``` is fine.

But I think those error messages are kind of pointless. It would be much better if the engine could do a health check on itself, and report a coordinated report on its incomplete pools, explaining the whole situation, after a reasonable time interval. It might be worth opening an issue at this point.

And finally, this behavior is inconsistent with what is done in ```StratEngine::initialize()``` where discovery of name conflict results in immediate error and termination of program. We should explain why that is in the comments, as there may be a perfectly good reason to behave differently, or else we should make the behavior consistent.

But, as I said, might as well wait until meeting before implementing any of this.

Reply:
In this case the pool really isn't incomplete, it's all there we just have a name conflict.  If we had some reporting mechanism in place or engine check we could certainly not log a message, but we have none of that so logging a message is better than silently doing nothing IMHO.

IMHO exiting the daemon is a really bad thing to do and should be avoided.  Once we do that the customer has no way to interact with stratis, the CLI won't work, we cannot send dbus signals or generate any kind of alerts and the health of whatever might be up and running cannot be monitored.  The daemon really needs to be running.

Reply:
> In this case the pool really isn't incomplete, it's all there we just have a name conflict. If we had some reporting mechanism in place or engine check we could certainly not log a message, but we have none of that so logging a message is better than silently doing nothing IMHO.

That is correct. I'm just maintaining that a reporting mechanism would be much better. Here you log if you have a name conflict, but of course you don't constantly log if you have an incomplete pool, because you expect that to change. But if the incomplete pool is never promoted to a complete pool, then that should be reported. But at this point we can't/don't. I'll file an issue.

Reply:
> IMHO exiting the daemon is a really bad thing to do and should be avoided. Once we do that the customer has no way to interact with stratis, the CLI won't work, we cannot send dbus signals or generate any kind of alerts and the health of whatever might be up and running cannot be monitored. The daemon really needs to be running.

So this is a reason to keep on going when, in the event loop, there is found to be a pool w/ a name conflict. That seems right to me. The question is, what about when the same situation arises in ```StratEngine::initialize```. Currently, the code exits. I'm saying that this _looks_ like an oversight, as I would expect the name conflict to be handled the same way in both places. If it is an oversight, than that code should be changed to be consistent. If it's deliberate, then a comment should be added to explain why the difference the way the same problem is handled between the two locations.